lib/openhab/core/dsl/actions.rb

```diff
@@ -10,9 +10,78 @@ module DSL
       # Module to import and steramlime access to OpenHAB actions
       #
       module Actions
-        #  def notify(user:, msg:); end
+        java_import org.openhab.core.library.types.PercentType
+        include Logging
 
-        #  def notifyAll(message); end
+        NotificationAction = nil # Avoid the (NameError) uninitialized constant error
+        OpenHAB::OSGI.services('org.openhab.core.model.script.engine.action.ActionService')&.each do |service|
+          java_import service.actionClass.to_s
+          logger.info("Loaded ACTION: #{service.actionClass}")
+        end
+        java_import org.openhab.core.model.script.actions.Exec
+        java_import org.openhab.core.model.script.actions.HTTP
+        java_import org.openhab.core.model.script.actions.Ping
+
+        #
+        # Return an OpenHAB Action object for the given scope and thing
+        #
+        # @param scope [String] The action scope
+        # @param thing_uid [String] Thing UID
+        #
+        # @return [Object] OpenHAB action
+        #
+        def action(scope, thing_uid)
+          $actions.get(scope, thing_uid)
+        end
+
+        #
+        # Send notification to an email or broadcast
+        #
+        # @param msg [String] The notification message to send
+        # @param email [String] The email address to send to. If nil, the message will be broadcasted
+        #
+        # @return [void]
+        #
+        def notify(msg, email: nil)
+          unless NotificationAction
+            raise NoMethodError, 'NotificationAction is not available. Please install the OpenHAB cloud addon'
+          end
+
+          if email
+            NotificationAction&.sendNotification email, msg
```

Comment:
You put the check above with a good error message.  This would now fail silently here with the &. is that what we want?  Or I guess the question I am asking is why use the &. here?

Reply:
The & is not necessary, but I think it makes no difference. I can remove it if you prefer?

Reply:
I think we should remove it.  The &. indicates we are Ok with the result being nil and I don't think we are here.  I think we want to fail.

Reply:
I'll remove the &, just waiting on the actions[][] and duration below